Description: Set GTK_IM_MODULE in Ubuntu on Xorg sessions
Origin: https://salsa.debian.org/input-method-team/im-config/-/commit/c3e8e161
Bug: https://bugs.debian.org/1031227
Bug-Ubuntu: https://launchpad.net/bugs/2007379

--- a/data/21_ibus.rc
+++ b/data/21_ibus.rc
@@ -4,6 +4,10 @@
 # hands off to set up IM if listed in $DESKTOP_SETUP_IBUS
 if find_match "$IM_CONFIG_CURRENT_DESKTOP" "$DESKTOP_SETUP_IBUS" ; then
     # ibus set up is owned by desktop --> do the same as "none"
+    if [ "$IM_CONFIG_PHASE" = 1 ] && [ "$XDG_SESSION_TYPE" = 'x11' ]; then
+        # exception for Xorg sessions; see https://bugs.debian.org/1031227
+        GTK_IM_MODULE=ibus
+    fi
     return
 fi
 
